pkg/cmd/roachtest/many_splits.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"context"
+	"fmt"
+)
+
+// runManySplits attempts to create 2000 tiny ranges on a 4-node cluster using
+// left-to-right splits and check the cluster is still live afterwards.
+func runManySplits(ctx context.Context, t *test, c *cluster) {
+	args := startArgs("--env=COCKROACH_SCAN_MAX_IDLE_TIME=5ms")
+	c.Put(ctx, cockroach, "./cockroach")
+	c.Start(ctx, t, args)
+
+	// Wait for upreplication then create many ranges.
+	func() {
+		db := c.Conn(ctx, 1)
+		defer db.Close()
+
+		waitForFullReplication(t, db)
+
+		const numRanges = 2000
```

Comment:
Are you sure about this "2000"

This will make the test run forever. Is the bug not visible with fewer ranges?

Reply:
```
20:58:47 many_splits.go:33: creating 2000 ranges...
21:00:01 many_splits.go:46: checking node 1 is live
```

It takes just over a minute now, how long is too long as far as roachtests go? It will show up with few ranges, so happy to reduce it as needed.